### DISCUSSION TITLE
redirect back after login

### DISCUSSION THREAD

**Discussion by Halvanhelv** [janko/rodauth-rails#241] [Q&A]
Hi! I used the settings to automatically redirect to past page and it works well for pages requiring authorization but for open pages it doesn't work http://rodauth.jeremyevans.net/rdoc/files/doc/guides/login_return_rdoc.html

then I did 

```
if rodauth.logged_in?
      # Return to the referrer page after multifactor authentication
      unless rodauth.two_factor_authenticated?
        session[rodauth.two_factor_auth_redirect_session_key] = request.referer
      end
    else
      session[rodauth.login_redirect_session_key] = request.referer
    end
```

and now the redirect works for open pages but not for authorized pages. how can I make the code work in both cases?

or what other approach should I take? Thanks

**Comment by janko** [comment]
I kind of gave up trying to make Rodauth return to the original page when it didn't require authentication, meaning when the user visited the login page directly. I'm not sure even Devise handles this scenario.

But I will take a closer look why the code above stops returning to pages that required authentication.

**Reply by janko** [reply]
OK, it makes sense that your attempt with `request.referer` didn't work, because it's not recording the last page before login, it's recording the one before the last.

The official instructions most likely don't work in MFA case because the original path stored internally by `two_factor_auth_return_to_requested_location? true` setting is being overwritten in the `route` block.

I don't have a robust solution for this, it seems to me like a difficult problem to solve 🤷🏻‍♂️

**Reply by Halvanhelv** [reply]
It’s a pity, of course, that this problem is deeper than it seemed, because such functionality is quite important, apparently we’ll have to look for some workarounds


Thanks for the answer
